### DISCUSSION TITLE
Add support for Bookmark Button for the SearchBar Component

### DISCUSSION THREAD

**Discussion by KevinvdBurg** [software-mansion/react-native-screens#1954] [Ideas]
We are currently implementing a feature that involves using headerSearchBarOptions within React Navigation. We want to utilize the bookmark button feature with a custom icon. However, upon inspection and attempted implementation, it appears that the react-native-screens library does not expose the bookmark button functionality of UISearchBar that is available in native iOS development as indicated by Apple's documentation and various discussions (e.g., on Stack Overflow).

**References**

[UISearchBar Apple Documentation](https://developer.apple.com/documentation/uikit/uisearchbar/1624298-showsbookmarkbutton)
[Customising Bookmark Icon Added to UISearchBar - Stack Overflow](https://stackoverflow.com/questions/14216219/customising-bookmark-icon-added-to-uisearchbar)

**Feature Description**

The specific feature request is to add the ability to:

Enable the bookmark button in the UISearchBar component within react-native-screens.
Provide the ability to customize the icon used for the bookmark button.

**Use Case**

This feature is essential for apps that require a bit more customization in their search experience, particularly when trying to maintain a consistent design language across the app. The ability to replace the default bookmark icon with a custom icon would allow for a more integrated and seamless user interface that feels native to iOS.

Thank you for considering this feature request.

**Comment by tboba** [comment]
Hi @KevinvdBurg, thanks for creating this thread!
Yep, as I can see we don't currently support showing bookmark button for the search bar on iOS, but it seems to be fairly easy to work on! We'll see what we can do 🙏

Cheers!
